src/tree/v2/ResourceTreeDataProviderBase.ts

```diff
@@ -63,8 +63,13 @@ export abstract class ResourceTreeDataProviderBase extends vscode.Disposable imp
 
     onDidChangeTreeData: vscode.Event<void | ResourceGroupsItem | ResourceGroupsItem[] | null | undefined> = this.onDidChangeTreeDataEmitter.event;
 
-    getTreeItem(element: ResourceGroupsItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
-        return element.getTreeItem();
+    async getTreeItem(element: ResourceGroupsItem): Promise<vscode.TreeItem> {
+        const treeItem = await element.getTreeItem();
+        treeItem.id = this.itemCache.getId(element);
+
+        // TODO: remove this when we're done working with ids
```

Comment:
Should this be done before or after this PR is merged?

Reply:
I would like to keep it around until reveal and find are merged. I've experienced issues where the cache is cleared in the middle of a `getId` call and the id returned is wrong. Although I've fixed this issue by caching the parent on the items, putting the id here makes debugging a lot easier in case I stumble upon a similar issue later on.